kroxylicious-operator/src/test/java/io/kroxylicious/operator/ProxyServiceTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.operator;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+
+import io.fabric8.kubernetes.api.model.Service;
+
+import io.kroxylicious.crapi.v1alpha1.KafkaProxy;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ProxyServiceTest {
+
+    public static final YAMLMapper YAML_MAPPER = new YAMLMapper()
+            .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER);
+
+    private static KafkaProxy kafkaProxyFromString(String yaml) throws JsonProcessingException {
+        // TODO should validate against the CRD schema, because the DependentResource
+        // should never see an invalid resource in production
+        return YAML_MAPPER.readValue(yaml, KafkaProxy.class);
+    }
+
+    @Test
+    void test() throws JsonProcessingException {
+        // Given
+        var kafkaProxy = kafkaProxyFromString("""
+                kind: KafkaProxy
+                apiVersion: kroxylicious.io/v1alpha1
+                metadata:
+                  name: my-example-proxy
+                spec:
+                  bootstrapServers: my-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092
+                """);
```

Comment:
Should a test provide the CR as a String literal? 
* One alternative would be to use a classpath resource. That makes the test a bit harder to read, but does mean you can potentially use tools like `yq` to bulk-update the test CRs (while the CRD schema is in flux, we might change the schema incompatibly and need to update all the test CR instances, which will be painful if they're string literals).
* Another alternative here is to use the generate java ± builders. That would again make the tests harder to read, but does mean that incompatible CRD schema changes will tend to present as compile time errors.

Preferences?